### PR TITLE
Added nore ignores. Removed.env example file

### PR DESCRIPTION
**Pull request by gaaarfild**
**Reasons for making this change:**

`.env.example` should not be ignored, because it contains exmplaes of actual parameters and on deploy, needs to be copied to `.env` file with appropriate parameter values.

**Links to documentation supporting these rule changes:** 

Didn't find a documentation, but I think this is common practice, to use such files.
https://mattstauffer.co/blog/laravel-5.0-environment-detection-and-environment-variables#introducing-php-dotenv

### CODE REVIEW

File: Laravel.gitignore [hunk 1]
```diff
@@ -1,5 +1,7 @@
 vendor/
 node_modules/
+.rocketeer/
```
Comment (shiftkey) [inline Laravel.gitignore]: These changes seem unrelated to the PR - can you tell me a bit about why they're useful here?

**Comment by shiftkey** [comment]
> Didn't find a documentation, but I think this is common practice, to use such files.

The section further down in [Customizing the example file](https://mattstauffer.co/blog/laravel-5.0-environment-detection-and-environment-variables#customizing-the-example-file) actually covers this nicely:

> That's it! Need 10 variables for each install? Add those 10 variables to your `.env.example` file with sensible (or silly) defaults. This file will get committed to your source control, and then **each new install can start out by running `cp .env.example .env` and then customizing `.env`**.

And you can't do this if the `.env.example` is in version control, so :thumbsup: to this change...

File: Laravel.gitignore (hunk 1 above)
Comment (gaaarfild) [inline Laravel.gitignore]: Maybe I don't understand the system, but afaik, we don't use different .gitignore files in one folder. So covering some deployment systems in this file is a good way. Or am I wrong?

Comment (shiftkey) [inline Laravel.gitignore]: @gaaarfild I'm fine with handling multiple changes to a file in this PR, I'd just love some context on what these folders represent - and links to the relevant projects in case others want to know more...

Comment (gaaarfild) [inline Laravel.gitignore]: Sorry for my rudeness. This is the package https://github.com/rocketeers/rocketeer

Comment (shiftkey) [inline Laravel.gitignore]: If these rules are related, could you move them out to their own section like this:

```
# Rocketeer - task runner and deployment package
.rocketeer/
.phpintel/
```

**Comment by shiftkey** [comment]
@gaaarfild fantastic, thanks for those changes!
